Static ACL support array of properties now

ulion opened the pull request:
to make the acl rule reusable for multiple properties. now the static acl in model json support array of string to setup acl in batch for group of properties.

slnode commented:
Can one of the admins verify this patch? To accept patch and trigger a build add comment ".*ok\W+to\W+test.*"

raymondfeng commented on common/models/acl.js:
```diff
@@ -256,10 +256,10 @@ module.exports = function(ACL) {
     if (modelClass && modelClass.settings.acls) {
       modelClass.settings.acls.forEach(function(acl) {
-        if (!acl.property || acl.property === ACL.ALL || property === acl.property) {
+        if (!acl.property || acl.property === ACL.ALL || property === acl.property || Array.isArray(acl.property) && acl.property.indexOf(property) >= 0) {
```
Please reformat the code so that it doesn't go beyond 80 chars per line.

raymondfeng commented:
Thanks for the patch. Please add a test case too.

ulion commented:
adjusted and added test code for it.

2015-03-06 0:33 GMT+08:00 Raymond Feng <notifications@github.com>:

> Thanks for the patch. Please add a test case too.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/strongloop/loopback/pull/1174#issuecomment-77397350>.
>



-- 
Ulion

ulion commented:
Thank you

raymondfeng commented:
@crandmck Can we add it to our docs? For the same ACL rule, now we support an array of values for the `property` attribute to match more than one methods. For example,
```json
"acls": [ {
  "principalType": "ROLE",
  "principalId": "$everyone",
  "property": ["create", "update"],
  "permission": "DENY"
  }
]
```

crandmck commented:
Added a mention to http://docs.strongloop.com/display/LB/Controlling+data+access and http://docs.strongloop.com/display/LB/Model+definition+JSON+file#ModeldefinitionJSONfile-ACLs
